backend/backend/src/main/java/de/aaaaaaah/velcom/backend/newaccess/committaccess/CommitReadAccess.java

```diff
@@ -258,6 +258,37 @@ public FullCommit getFullCommit(RepoId repoId, CommitHash commitHash)
 		}
 	}
 
+	/**
+	 * @param root the root commit
+	 * @return all tracked commits descending from the given root.
+	 */
+	public List<CommitHash> getDescendantCommits(Commit root) {
+		String query = "WITH RECURSIVE rec(hash) AS (\n"
+			+ "  VALUES (?)\n"                                    // <-- Binding #1 - Root hash
+			+ "\n"
+			+ "  UNION\n"
+			+ "\n"
+			+ "  SELECT commit_relationship.child_hash\n"
+			+ "  FROM commit_relationship\n"
+			+ "  JOIN rec\n"
+			+ "  ON rec.hash = commit_relationship.parent_hash\n"
+			+ "  WHERE repo_id = ?\n"                             // <-- Binding #2 - repo id
+			+ ")\n"
+			+ "\n"
+			+ "SELECT rec.hash "
+			+ " FROM rec\n"
+			+ "JOIN known_commit ON rec.hash = known_commit.hash\n"
+			+ "WHERE known_commit.tracked;";
```

Comment:
For consistency with the other recursive queries, these changes would be nice:
- Indent the `\n` around `UNION`
- Always put `ON`s on a new line and indent them
- Put newline after `SELECT rec.hash`
- Indentation is always 2 spaces (see `" FROM rec\n"`)

And set the last lines to
```java
+ "WHERE known_commit.tracked\n"
+ "";
```